src/DaPigGuy/libPiggyEconomy/providers/BedrockEconomyProvider.php

```diff
@@ -24,32 +24,48 @@ public static function checkDependencies(): bool
 
     public function __construct()
     {
-        $this->api = BedrockEconomyAPI::legacy();
-        $this->currency = BedrockEconomy::getInstance()->getCurrencyManager();
+        $this->api = BedrockEconomyAPI::ASYNC();
```

Comment:
The callback API provided by BedrockEconomy may be easier to work with for contributors to this provider in the future

Reply:
Not sure if you were talking about the `ClosureAPI`. Let me know if you meant something else.
Using this API also fixes the virion and deprecated parameter issue.